soc/ite/ec/it8xxx2/kernel_no_lto.cmake

```diff
@@ -0,0 +1,38 @@
+set(KERNEL_CMAKE "${ZEPHYR_BASE}/kernel/CMakeLists.txt")
+set(MARKER "# AUTO_INJECTED_NO_LTO")
+
+execute_process(
+  COMMAND ${CMAKE_COMMAND} -E env python3
+          ${CMAKE_CURRENT_LIST_DIR}/cleanup_injected_code.py
+          ${KERNEL_CMAKE}
+  RESULT_VARIABLE CLEAN_RESULT
+  OUTPUT_QUIET
+  ERROR_QUIET
+)
+
+if(NOT CLEAN_RESULT EQUAL 0)
+  message(WARNING "[no-LTO] Warning: cleanup script encountered an issue, continuing anyway.")
+endif()
+
+# Prepare the code snippet to inject
+message(STATUS "[no-LTO] Injecting no-LTO configuration into kernel/CMakeLists.txt...")
+set(INJECT_CODE "${MARKER}
+set_source_files_properties(busy_wait.c device.c kheap.c idle.c mutex.c sem.c work.c
+                            thread.c sched.c timeout.c time.c poll.c events.c banner.c
+                            system_work_q.c
+                            PROPERTIES COMPILE_FLAGS \"-fno-lto -g\")")
+
+# Append the injected code to the end of the file
+file(APPEND ${KERNEL_CMAKE} "${INJECT_CODE}")
```

Comment:
uh, why is this needed? The whole append/remove thing feels a bit hacky to be honest, hope we can find a different solution, you just need the `-fno-lto -g` in the build command line for the files above right?

Reply:
The reason is that setting `-fno-lto` elsewhere (e.g. board-level CMake) didn’t work likely because of build order issues. Kernel sources were still built with LTO, so I had to apply it directly here to make sure those files compile without LTO and can reside in RAM.

Reply:
ok so this is basically touching the source file to somehow convince cmake to build it again with the new flag?

Reply:
Yes, that's correct, it temporarily modifies the kernel CMake file to ensure those specific sources are rebuilt with `-fno-lto`. After the build configuration is complete, the injected changes are automatically cleaned up to restore the original file state.